src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -557,33 +558,69 @@ private HttpHandlerDiagnosticListener() : base(DiagnosticListenerName)
 
         private void RaiseRequestEvent(HttpWebRequest request)
         {
-            // If System.Net.Http.Request is on, raise the event
-            if (this.IsEnabled(RequestWriteName))
+            if (request.Headers.Get(RequestIdHeaderName) != null)
             {
-                long timestamp = Stopwatch.GetTimestamp();
-                this.Write(RequestWriteName,
-                    new
+                // this request was instrumented by previous RaiseRequestEvent
+                return;
+            }
+
+            if (this.IsEnabled(ActivityName, request))
+            {
+                var activity = new Activity(ActivityName);
+
+                // Only send start event to users who subscribed for it, but start activity anyway
+                if (this.IsEnabled(RequestStartName))
+                {
+                    long timestamp = Stopwatch.GetTimestamp();
```

Comment:
Bummer.  I suppose the breakage of Activity.Current is not that surprising (This code predates Tasks).   

While fixing Activity Timestamping on .NET Desktop is probably still worth doing (for those cases where it does work) I agree that it is not something that is helpful in this instance.  We can table my request.   

I am still a bit conflicted about Stopwatch.TimeStamp because the listener could have done the Stopwatch.GetTimestamp() call itself.   Moreover I don't really like using stopwatch ticks in things that are 'public' (someone has to convert them sooner or later). 

My preference is to simply remove the timestamping (allow the listener to do it if they care), but if doing that will cause any grief (e.g. there is already code that depends on it), then I am OK with simply leaving it (all this code is roughly a hack that will go away 'eventually').  Let me know...